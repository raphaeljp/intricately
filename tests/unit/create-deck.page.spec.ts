import { shallowMount } from '@vue/test-utils';
import NewDeck from '@/views/deck/create-deck.page.vue';
import CardInput from '@/components/card-input.component.vue';

const factory = (values = {}) => shallowMount(NewDeck, {
  data() { return { ...values }; },
});

describe('Create deck page', () => {
  it('should be mounted', () => {
    const wrapper = factory();
    expect(wrapper).toBeTruthy();
  });

  it('should have CardInput Component', () => {
    const wrapper = factory();
    const inputs = wrapper.getComponent(CardInput);
    expect(inputs).toBeTruthy();
  });

  it('should have 11 inputs on page', () => {
    const wrapper = factory();
    const inputs = wrapper.findAllComponents(CardInput);
    expect(inputs).toHaveLength(11);
  });

  test('renders correctly', () => {
    const wrapper = factory();
    expect(wrapper).toMatchSnapshot();
  });
});
