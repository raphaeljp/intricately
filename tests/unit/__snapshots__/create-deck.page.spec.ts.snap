// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create deck page renders correctly 1`] = `
<div id="new-deck__page">
  <form id="deck-of-cards__form">
    <div id="form__inputs">
      <cardinput-stub id="0"></cardinput-stub>
      <cardinput-stub id="1"></cardinput-stub>
      <cardinput-stub id="2"></cardinput-stub>
      <cardinput-stub id="3"></cardinput-stub>
      <cardinput-stub id="4"></cardinput-stub>
      <cardinput-stub id="5"></cardinput-stub>
      <cardinput-stub id="6"></cardinput-stub>
      <cardinput-stub id="7"></cardinput-stub>
      <cardinput-stub id="8"></cardinput-stub>
      <cardinput-stub id="9"></cardinput-stub>
    </div>
    <div id="rotation__card">
      <cardinput-stub id="rotation" value="" hidelabel="true" nomargin="true"></cardinput-stub>
    </div>
    <div id="form__button"><button>Submit</button></div>
  </form>
</div>
`;
